### DISCUSSION TITLE
[RFC] Allow protected properties in PropertyTypeFromStrictSetterGetterRector

### DISCUSSION THREAD

**Discussion by buffcode** [rectorphp/rector#8431] [Ideas]
Currently `PropertyTypeFromStrictSetterGetterRector` only allows `private` untyped properties, so the overall logic can ensure that only this class reads/writes to the attribute with a given type. 👍 

In our codebase we have lots of `protected` properties, because historically Doctrine/JMS/... had serialization problems with `private` properties.

If accepted, I would like to raise a PR that - by default - only allows `private` properties (as currently), but allows to configure the rector with different allowed flags:
```php
$rectorConfig->ruleWithConfiguration(
    \Rector\TypeDeclaration\Rector\Class_\PropertyTypeFromStrictSetterGetterRector::class,
    [
        \Rector\TypeDeclaration\Rector\Class_\PropertyTypeFromStrictSetterGetterRector::ALLOWED_MODIFIERS => [
            \PhpParser\Node\Stmt\Class_::MODIFIER_PRIVATE,
            \PhpParser\Node\Stmt\Class_::MODIFIER_PROTECTED,
        ]
    ]
);
```

Thanks for your feedback!

This would allow eg. for one-off modification of the code-base.

**Comment by samsonasik** [comment]
You may looking on configurable INLINE_PUBLIC on `TypedPropertyFromAssignsRector` for that, see

https://github.com/rectorphp/rector-src/blob/f47b38ed3980eb79113bb1b3b1289fb5a5c05c0f/rules/TypeDeclaration/Rector/Property/TypedPropertyFromAssignsRector.php#L37-L51
